proto/spire-next/types/entry.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+package spire.types;
+option go_package = "github.com/spiffe/spire/proto/spire-next/types";
+
+import "spire-next/types/selector.proto";
+import "spire-next/types/spiffeid.proto";
+
+message Entry {
+    // Globally unique ID for the entry.
+    string id = 1;
+
+    // The SPIFFE ID of the identity described by this entry.
+    spire.types.SPIFFEID spiffe_id = 2;
+
+    // Who the entry is delegated to. If the entry describes a node, this is
+    // set to the SPIFFE ID of the SPIRE server of the trust domain (e.g.
+    // spiffe://example.org/spire/server). Otherwise, it will be set to a node
+    // SPIFFE ID.
+    spire.types.SPIFFEID parent_id = 3;
+
+    // The selectors which identify which entities match this entry. If this is
+    // an entry for a node, these selectors represent selectors produced by
+    // node attestation. Otherwise, these selectors represent those produced by
+    // workload attestation.
+    repeated spire.types.Selector selectors = 4;
+
+    // The time to live for identities issued for this entry.
+    int32 ttl = 5;
+
+    // A list of trust domains the identity described by this entry federates
```

Comment:
nit: it would be good to clarify that is the SPIFFE ID of the trust domain

Reply:
That is true for the existing entry message, but doesn't necessarily have to be the case going forward. We'd obviously have to use the SPIFFE ID internally, but from an API perspective, which is the more convenient? Also, if this is a string, then the server could presumably accept either (e.g. by using `spiffeid.TrustDomainFromString`).

If we decide this must be a SPIFFE ID, I'd advocate for changing it to `repeated spire.types.SPIFFEID`

Reply:
I see now. It's fair to keep the comment as is then.